nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PutSlack.java

```diff
@@ -257,16 +272,27 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             jsonWriter.close();
 
             final URL url = URI.create(context.getProperty(WEBHOOK_URL).evaluateAttributeExpressions(flowFile).getValue()).toURL();
-            final HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-            conn.setRequestMethod("POST");
-            conn.setDoOutput(true);
+            int responseCode;
+
+            if (sslService != null) {
+                final HttpsURLConnection conn = (HttpsURLConnection) url.openConnection();
```

Comment:
`HttpsURLConnection` is a subclass of `HttpURLConnection`, so the approach should be restructured to remove the duplicative elements and scope the condition to setting the SSLSocketFactory when the SSL Context Service is configured.